RFC for GitHub issue labels

m-allanson opened the pull request:
[View formatted rfc](https://github.com/gatsbyjs/rfcs/blob/github-issue-labels/text/0000-github-issue-labels.md)

calcsam commented:
We definitely should do this. Culling labels down from 20 to 9 will be a huge step towards usability and a triage system.

shannonbux commented:
I like this although currently I use the doc and ux labels to pull things into the projects (it's easy to search for those labels and pull all labelled things in).

Without those labels, I'd need a new process for pulling community stuff into those issues and I'm open to alternatives (actually, currently *so* many things have the documentation label, I haven't been able to address all issues yet). Any ideas?

fk commented:
Hey @m-allanson! Thanks for putting this together, and also @jlengstorf for https://github.com/gatsbyjs/gatsby/issues/4311!

I've yet to make up my mind about the details — quick thoughts for now:

* Initially I was leaning towards "more labels" – for example I was missing `question` in this RFC and didn't find `discussion` clear enough as an umbrella label. @jlengstorf makes a clear distinction between the two that I can follow. I also saw that Mike explicitly speaks of "rename to type: `discussion` or _similar_", and spent a few minutes thinking about _something similar_…🙃😬 Thinking out loud, I can see questions transitioning to `documentation`, `discussion` or `feature` quite fast, too. Also the label description (and a section in CONTRIBUTING.md, e.g. like https://github.com/twbs/bootstrap/blob/v4-dev/.github/CONTRIBUTING.md#issues-and-labels) could help  …still so hard to let go ;-)
* That said, I'm all for making things simpler and esp. reducing labels. Btw. Mike, is it a coincidence that the total amount of 11 labels that you suggest would result in a non-scrolling popover? ;-)
![image](https://user-images.githubusercontent.com/21834/38042933-f9e3d96c-32b5-11e8-935c-7f59c52b8cfd.png)
* Maybe Jason's suggested `status: blocked` is a good addition that also covers `upstream-issue`?
* OK, now I'm flirting with "more labels" again 😁😅…
* I would love to see definitions for each label à la @jlengstorf — IMO this should help getting things settled, and probably also gets us ~90% to label descriptions and a possible CONTRIBUTING.md addition on issues.

fk commented:
@shannonbux When writing

> Without those labels, I'd need a new process for pulling community stuff into those issues

by "issues", do you mean the documentation and UX "project"s by any chance?

shannonbux commented:
@fk haha yes, projects! Whoops!

On Wed, Mar 28, 2018 at 12:25 PM, Florian Kissling <notifications@github.com
> wrote:

> @shannonbux <https://github.com/shannonbux> When writing
>
> Without those labels, I'd need a new process for pulling community stuff
> into those issues
>
> by "issues", do you mean the documentation and UX "project"s by any chance?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gatsbyjs/rfcs/pull/1#issuecomment-376987702>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/Ae9o2lXoU6Vt6xB8orC4VBZeFuJyQya3ks5ti9WRgaJpZM4SzZgm>
> .
>

KyleAMathews commented on text/0000-github-issue-labels.md:
```diff
@@ -0,0 +1,123 @@
+| Feature Request | ✅keep | rename to `type: feature` or similar |
+| good first issue | ✅keep | exact name is used by GitHub, don't rename |
+| GraphQL | ❌delete | too specific (currently applied to 13 issues) |
+| Hacktoberfest | ✅keep | exact name is used by Hacktoberfest, don't rename  |
```
I wonder if there's a way to automatically add this tag to ones we tag with "good first issue"? Otherwise we'd need to manually keep the two in sync.

m-allanson replied:
Yeah I think this would be a good next step. Maybe using the existing probot or something like [Danger JS](http://danger.systems/js/).

m-allanson commented:
> Without those labels, I'd need a new process for pulling community stuff into those projects and I'm open to alternatives (actually, currently so many things have the documentation label, I haven't been able to address all issues yet). Any ideas?

@shannonbux Huh, reviewing my post it looks like I've been inconsistent with the docs label, I'll update that. I think we should keep the Documentation label (as `type: documentation`), but remove the UX one.

Instead of labelling an issue as UX, you'd add it straight to the UX project. The issue would probably have a `type: documentation` or `type: feature` label.

I'm very open to rethinking if the above isn't a useful workflow, we should make this labelling change useful for everyone.

> Initially I was leaning towards "more labels" – for example I was missing question in this RFC and didn't find discussion clear enough as an umbrella label. @jlengstorf makes a clear distinction between the two that I can follow.

My initial list may have been too minimal - a `type: question` label in addition to `type: discussion` sounds good.

...After writing descriptions for each label (see below), I'm not sure if the difference between _question_ and _discussion_ is distinct enough. It should always be easy to choose a label and I think the line between these two can get blurred quite easily. I guess we're after a 'free form' label that's not focussed on one particular thing. How does `type: question or discussion` sound? (It's not a particularly imaginative label name)

> Also the label description (and a section in CONTRIBUTING.md, e.g. like https://github.com/twbs/bootstrap/blob/v4-dev/.github/CONTRIBUTING.md#issues-and-labels) could help

Agreed! I've added a first draft at descriptions below.

> Thinking out loud, I can see questions transitioning to documentation, discussion or feature quite fast

Yeah I think that's ok and to be encouraged.


> is it a coincidence that the total amount of 11 labels that you suggest would result in a non-scrolling popover? ;-)

Hah! That one _is_ a coincidence :)


> Maybe Jason's suggested status: blocked is a good addition that also covers upstream-issue

That could be good too. Thinking of other reasons that issues could be stalled - maybe `status: needs repro` could be made more generic by changing it to `status: needs more info` (aka @jlengstorf's `status: needs clarification` label)?

> I would love to see definitions for each label

Good idea. I've added a first run of definitions below.

### Summary

Here's a summary of all the label names with descriptions.

I've added `status: blocked` and `status: needs more info` labels, along with `type: question or discussion`. I've also added emojis to denote the label categories (taken from @jlengtorf's initial labelling issue https://github.com/gatsbyjs/gatsby/issues/4311#issue-301482647)

| Name | Description |
|---|---|
| good first issue | Issue that doesn't require previous experience with Gatsby |
| hacktober | Issue that doesn't require previous experience with Gatsby |
| 🏷 type: bug | Issue that describes a bug in Gatsby |
| 🏷 type: documentation | Issue for improving or updating Gatsby's documentation |
| 🏷 type: feature | Issue asking for a new feature to be added, or an existing one to be extended or modified. |
| 🏷 type: maintenance | Issue describing a change that isn't a bug, feature or documentation change. Things like refactoring, build tasks, CI changes, dependency updates or additional tests. |
| 🏷 type: question or discussion | Issue discussing or asking a question about Gatsby |
| 📍 status: blocked | Issue that's blocked by factors outside Gatsby |
| 📍 status: help wanted | Issue we need or would love help from the community to resolve |
| 📍 status: needs more info | Issue that needs triaging and reproducible examples or more information to be resolved |
| 🤖 bot: review | An automated label added by a bot. |
| 🤖 bot: stale? | An automated label added by a bot. Obsolete question or discussion that has either ended or concerns an older version |

KyleAMathews commented:
This RFC has entered the final comment period.

KyleAMathews commented:
And it's in! Thanks @m-allanson for the first RFC :-D

m-allanson commented:
🎉 I'll implement this over the next few days.
